src/daemon.rs

```diff
@@ -65,13 +86,59 @@ impl Daemon {
         cas: crate::cas::ContentAddressable,
     ) -> Result<(), ExitError> {
         let (activity_tx, activity_rx) = chan::unbounded();
-        let server = rpc::Server::new(socket_path, activity_tx)?;
+
+        let server = rpc::Server::new(socket_path, activity_tx, self.build_events_tx.clone())?;
         let mut pool = crate::thread::Pool::new();
         pool.spawn("accept-loop", move || {
             server
                 .serve()
                 .expect("failed to serve daemon server endpoint");
         })?;
+        let build_events_rx = self.build_events_rx.clone();
+        let mon_tx = self.mon_tx.clone();
+        pool.spawn("build-loop", move || {
+            let mut project_states: HashMap<NixFile, Event> = HashMap::new();
+            let mut event_listeners: Vec<chan::Sender<Event>> = Vec::new();
+
+            for msg in build_events_rx {
+                debug!("Build loop message"; "message" => ?msg);
+                mon_tx
+                    .send(msg.clone())
+                    .expect("listener still to be there");
+                match &msg {
+                    LoopHandlerEvent::BuildEvent(ev) => match ev {
+                        Event::SectionEnd => (),
+                        Event::Started { nix_file, .. }
+                        | Event::Completed { nix_file, .. }
+                        | Event::Failure { nix_file, .. } => {
+                            project_states.insert(nix_file.clone(), ev.clone());
+                            event_listeners.retain(|tx| {
+                                let keep = tx.send(ev.clone()).is_ok();
+                                debug!("Sending"; "event" => format!("{:#?}", ev), "keep" => keep);
```

Comment:
```suggestion
                                debug!("Sent"; "event" => ?ev, "keep" => keep);
```

See https://docs.rs/slog/2.5.2/slog/macro.log.html#fmtdisplay-and-fmtdebug-values.